### PR TITLE
Clean up Kyuubi Hive JDBC client

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Code clean up

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
Thanks, merging to master

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java [hunk 1]
```diff
@@ -353,32 +361,4 @@ static int columnScale(int columnType, JdbcColumnAttributes columnAttributes)
         throw new KyuubiSQLException("Invalid column type: " + columnType);
     }
   }
-
-  public Integer getNumPrecRadix() {
```
Comment (yaooqinn) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java:357]: why delete this public API

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiQueryResultSet.java [hunk 2]
```diff
@@ -318,7 +313,7 @@ public boolean next() throws SQLException {
       return false;
     }
 
-    /**
+    /*
```
Comment (yaooqinn) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiQueryResultSet.java:316]: nit

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/cli/ColumnBasedSet.java [hunk 3]
```diff
@@ -112,43 +75,11 @@ public int numRows() {
     return columns.isEmpty() ? 0 : columns.get(0).size();
   }
 
-  @Override
-  public ColumnBasedSet extractSubset(int maxRows) {
```
Comment (yaooqinn) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/cli/ColumnBasedSet.java:116]: this is a public programing api...

**Comment by yaooqinn** [comment]
better to let others know what you cleaned, not just `code cleanup`

**Comment by pan3793** [comment]
The main purpose of the "code cleanup" is to remove un-referenced code and break nothing, some data structures are shared by client and server, this PR removed the server-specific APIs and code as well.

I'm focused on JDBC standard API, and you are right, some client-side public programing APIs were removed, let me send another PR to restore them.

**Comment by yaooqinn** [comment]
let's make the PR title and description as clear as possible, especially in the role of Kyuubi PPMC.

**Pull request closed without merging**